test/rpc/rpc-handlerSpec.ts

```diff
@@ -0,0 +1,344 @@
+import { EVENT } from '../../src/constants'
+import { TOPIC, RPC_ACTIONS, RPCMessage } from '../../binary-protocol/src/message-constants'
+
+import { DefaultOptions, Options } from '../../src/client-options'
+import { RPCHandler, RPCProvider } from '../../src/rpc/rpc-handler'
+import { RPCResponse } from '../../src/rpc/rpc-response'
+import { TimeoutRegistry } from '../../src/util/timeout-registry'
+
+describe('RPC handler', () => {
+  let services: any
+  let rpcHandler: RPCHandler
+  let handle: Function
+  let rpcProviderSpy: sinon.SinonSpy
+  let data: any
+  const name = 'myRpc'
+  const rpcAcceptTimeout = 3
+  const rpcResponseTimeout = 10
+  const options = Object.assign({}, DefaultOptions, { rpcAcceptTimeout, rpcResponseTimeout })
+
+  beforeEach(() => {
+    services = getServicesMock()
+    rpcHandler = new RPCHandler(services, options)
+    handle = services.getHandle()
+    rpcProviderSpy = sinon.spy()
+    data = { foo: 'bar' }
+  })
+
+  afterEach(() => {
+    services.connectionMock.verify()
+    services.timeoutRegistryMock.verify()
+    services.loggerMock.verify()
+  })
+
+  it('validates parameters on provide, unprovide and make', () => {
+    expect(rpcHandler.provide.bind(rpcHandler, '', () => {})).to.throw()
+    expect(rpcHandler.provide.bind(rpcHandler, 123, () => {})).to.throw()
+    expect(rpcHandler.provide.bind(rpcHandler, null, () => {})).to.throw()
+
+    expect(rpcHandler.provide.bind(rpcHandler, name, null)).to.throw()
+    expect(rpcHandler.provide.bind(rpcHandler, name, 123)).to.throw()
+
+    expect(rpcHandler.unprovide.bind(rpcHandler, '')).to.throw()
+    expect(rpcHandler.unprovide.bind(rpcHandler, 123)).to.throw()
+    expect(rpcHandler.unprovide.bind(rpcHandler, null)).to.throw()
+    expect(rpcHandler.unprovide.bind(rpcHandler)).to.throw()
+
+    expect(rpcHandler.make.bind(rpcHandler, '')).to.throw()
+    expect(rpcHandler.make.bind(rpcHandler, 123)).to.throw()
+    expect(rpcHandler.make.bind(rpcHandler, null)).to.throw()
+    expect(rpcHandler.make.bind(rpcHandler)).to.throw()
+  })
+
+  it('registers a provider', () => {
+    const message = {
+      topic: TOPIC.RPC,
+      action: RPC_ACTIONS.PROVIDE,
+      name
+    }
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+      .withExactArgs(message)
+    services.timeoutRegistryMock
+      .expects('add')
+      .once()
+      .withExactArgs({ message })
+
+    rpcHandler.provide(name, rpcProviderSpy as RPCProvider)
+
+    sinon.assert.notCalled(rpcProviderSpy)
+  })
+
+  it('sends rpc request message on make', () => {
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+      .withExactArgs({
+        topic: TOPIC.RPC,
+        action: RPC_ACTIONS.REQUEST,
+        name,
+        parsedData: data,
+        correlationId: sinon.match.any
+    })
+
+    rpcHandler.make(name, data, () => {})
+  })
+
+  it('returns promise on make when no callback is passed', () => {
+    services.connectionMock
```

Comment:
should test the promise itself.. not just see it returns one

Reply:
This test hasn’t changed yet.

Reply:
The promises for all make requests are tested in the 'when making' describe block. I think it doesn't make sense to keep this since all the tests in 'when making' will fail if the returning object is not a promise.